server/src/main/java/org/elasticsearch/index/mapper/SourceValueFetcher.java

```diff
@@ -39,6 +39,11 @@ public SourceValueFetcher(String fieldName, SearchExecutionContext context) {
      */
     public SourceValueFetcher(String fieldName, SearchExecutionContext context, Object nullValue) {
         this(context.sourcePath(fieldName), nullValue);
+        if (context.isSourceEnabled() == false) {
```

Comment:
can this still happen with the changes you made?

Reply:
A custom MappedFieldType in a plugin could still be using SourceValueFetcher, so yes this could still happen.

Reply:
ok next question: do we have a test for this scenario? :)

Reply:
So, I can confirm a few things:
1. This still happens - the `text` field mapper hits it, for example. As would any other field mappers that use source only.
2. We have a test for it - it's the one with you commented on in 330_fetch_fields.
3. I'm not super happy to have this validation in the ctor. It feels a little sneaky. It's not a totally unreasonable place given where we are but it still feels a little sneaky to me.